### PR TITLE
Themes: Add Builder Referral Banner

### PR DESCRIPTION
**Pull request by jancavan**
#### Changes proposed in this Pull Request

This PR adds a banner to `/themes` that link to the Upwork referral landing page.

image

image

#### Testing instructions

1. Ensure that you are in the test. Set variation (builderReferralBanner): In console run `localStorage.setItem( 'ABTests', '{"builderReferralThemesBanner_20181218":"builderReferralBanner"}' );`
2. Go to `/themes`.
3. Check if you see the banner above the Search bar and make sure it links to the Upwork referral landing page.
4. Make sure that dismiss button closes banner

#### Testing Tracks Events

You can turn on analytics logging by running this in the console: localStorage.setItem( 'debug', 'calypso:analytics:*' )

- Refresh after running code to active analytics logging
- On viewing the banner: `calypso:analytics:tracks Record event "calypso_upwork_themes_banner_view" called with props {site_type: "jetpack"}`
- On clicking CTA: `calypso:analytics:tracks Recording event "calypso_upwork_themes_banner_start_now_button_click" with actual props {plan: "value_bundle", site_type: "wpcom", environment: "development", environment_id: "development", site_count: 291, …}`
- On dismissing: `calypso:analytics:tracks Recording event "calypso_upwork_themes_banner_dismiss_icon_click" with actual props {plan: "value_bundle", site_type: "wpcom", environment: "development", environment_id: "development", site_count: 291, …}`

Related: https://github.com/Automattic/wp-calypso/pull/29570

### CODE REVIEW

**Comment by jancavan** [comment]
@belcherj I think we'll be rolling this banner out to all plans, not just Business, just like the current banners that are on there. I'm clarifying with @pcdonohue and @p3ob7o 

We also need to incorporate the necessary Tracks events in this PR.

File: client/my-sites/themes/themes-banner/upwork.jsx [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	}
+
+	recordView() {
+		if ( this.props.isBannerVisible ) {
+			this.props.recordTracksEvent( 'calypso_upwork_themes_banner_view' );
+		}
+	}
+
+	recordClick = eventName => {
+		const plan = this.props.plan ? this.props.plan.productSlug : '';
```
Comment (cbauerman) [inline client/my-sites/themes/themes-banner/upwork.jsx]: looks like this should be `this.props.currentPlan`. I am only getting blank strings in the tracks events
```suggestion
		const plan = this.props.currentPlan ? this.props.currentPlan.productSlug : '';
```

**Comment by robertf4** [comment]
Code looks good and everything worked in testing. My only minor comment is that since the other actions/reducers/selectors in `client/state/themes/themes-ui` all have unit tests, it is probably best to add similar tests for the new upwork code.

**Comment by stephanethomas** [comment]
Changes look good to me. Feel free to merge it once you have addressed Robert's concern.

**Pull request merged**